xarray/core/options.py

```diff
@@ -41,6 +69,21 @@ def __init__(self, **kwargs):
             raise ValueError('argument names %r are not in the set of valid '
                              'options %r' % (invalid_options, set(OPTIONS)))
         self.old = OPTIONS.copy()
+
+        if 'io_engines' in kwargs:
```

Comment:
This clause needs test coverage -- somewhere in https://github.com/pydata/xarray/blob/master/xarray/tests/test_options.py